examples/scene-example.ts

```diff
@@ -0,0 +1,378 @@
+     * @todo comment
+     * @param frameNumber - for intermediate frames in accumulation rendering.
+     */
+    protected onFrame(frameNumber: number): void {
+        const gl = this._context.gl;
+
+        this._camera.viewport = [this._frameSize[0], this._frameSize[1]];
+
+        gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT);
+
+        gl.enable(gl.CULL_FACE);
+        gl.cullFace(gl.BACK);
+        gl.enable(gl.DEPTH_TEST);
+
+        this._forwardPass.frame();
+
+        gl.cullFace(gl.BACK);
+        gl.disable(gl.CULL_FACE);
+    }
+
+    /**
+     * @todo comment ...
+     */
+    protected onSwap(): void {
+    }
+
+
+    /**
+     *  @todo comment
+     */
+    protected generateScene(): void {
+        const gl = this._context.gl;
+
+        /* Create scene */
+        this._scene = new SceneNode('root');
+
+        /* Create mesh rendering program. */
+        const vert = new Shader(this._context, gl.VERTEX_SHADER, 'mesh.vert');
+        vert.initialize(require('./data/mesh.vert'));
+        const frag = new Shader(this._context, gl.FRAGMENT_SHADER, 'mesh.frag');
+        frag.initialize(require('./data/mesh.frag'));
+        const program = new Program(this._context, 'MeshProgram');
+        program.initialize([vert, frag]);
+
+        this.generateSphere1Node(this._scene, program);
+        this.generateSphere2Node(this._scene, program);
+        this.generateBoxNode(this._scene, program);
+    }
+
+    protected generateSphere1Node(parent: SceneNode, program: Program): SceneNode {
+        const gl = this._context.gl;
+
+        /* Create node with a mesh */
+        const node = parent.addNode(new SceneNode('mesh'));
+        const translate = mat4.fromTranslation(mat4.create(), vec3.fromValues(0.0, 0.0, 0.0));
+        const scale = mat4.fromScaling(mat4.create(), vec3.fromValues(0.4, 0.4, 0.4));
+        const transform = mat4.multiply(mat4.create(), translate, scale);
+        node.transform = transform;
+
+        /* Create and load texture. */
+        const texture = new Texture2D(this._context, 'Texture');
+        texture.initialize(128, 128, gl.RGB, gl.RGB, gl.UNSIGNED_BYTE);
+        texture.fetch('./data/logo.png', false).then(() => {
+            this.invalidate(true);
+        });
+
+        /* Create material */
+        const material = new SceneExampleMaterial('ExampleMaterial1', program);
+        material.texture = texture;
+        material.textured = true;
+
+        /* Create geometry. */
+        const geometry = new Sphere(
+            this._context,
+            'mesh',
+            1.0,
+            true);
+
+        const aMeshVertex = program.attribute('a_vertex', 0);
+        const aMeshTexCoord = program.attribute('a_texcoord', 1);
+        geometry.initialize(aMeshVertex, aMeshTexCoord);
+
+        const sphereComponent = new GeometryComponent();
+        sphereComponent.geometry = geometry;
+        sphereComponent.material = material;
+
+        node.addComponent(sphereComponent);
```

Comment:
What is the target domain of `Component`? For the domains of a scene graph, a renderer, or even geometry, I think this term is too unspecific.